src/core/qgsvectorfilewriter.cpp

```diff
@@ -703,6 +703,18 @@ void QgsVectorFileWriter::init( QString vectorFileName,
             break;
           }
 
+          case QVariant::Map:
+            // handle GPKG conversion to JSON
+            if ( mOgrDriverName == QLatin1String( "GPKG" ) )
```

Comment:
you could make it more general by checking the "JSON" string in GDALGetMetadata(GDALGetDriverByName(mOgrDriverName), DMD_CREATIONFIELDDATASUBTYPES, nullptr)

Reply:
done